pkg/upload/android-proguard.go

```diff
@@ -1,128 +1,149 @@
-	Path            utils.UploadPaths `arg:"" name:"path" help:"(required) Path to directory or file to upload" type:"path"`
-	VersionCode     string            `help:"Module version code"`
-	VersionName     string            `help:"Module version name"`
+	ApplicationId string            `help:"Module application identifier"`
+	AppManifest   string            `help:"Path to app manifest file" type:"path"`
+	BuildUuid     string            `help:"Module Build UUID"`
+	Path          utils.UploadPaths `arg:"" name:"path" help:"Path to directory or file to upload" type:"path" default:"."`
+	Variant       string            `help:"Build type, like 'debug' or 'release'"`
+	VersionCode   string            `help:"Module version code"`
+	VersionName   string            `help:"Module version name"`
 }
 
-func ProcessAndroidProguard(paths []string, applicationId string, appManifestPath string, buildUuid string, configuration string, versionCode string, versionName string, endpoint string, timeout int, retries int, overwrite bool, apiKey string, failOnUploadError bool, dryRun bool) error {
+func ProcessAndroidProguard(apiKey string, applicationId string, appManifestPath string, buildUuid string, paths []string, variant string, versionCode string, versionName string, endpoint string, retries int, timeout int, overwrite bool, dryRun bool) error {
 
-	uploadFileOptions := make(map[string]map[string]string)
+	var mappingFile string
+	var err error
 
 	for _, path := range paths {
-		// Path is a directory
 		if utils.IsDir(path) {
-			mergedManifestsPath := filepath.Join(path, "build", "intermediates", "merged_manifests")
-			if utils.IsDir(mergedManifestsPath) {
-				variants, err := android.BuildVariantsList(mergedManifestsPath)
-				if err != nil {
-					log.Error(err.Error(), 1)
-				}
 
-				for _, variant := range variants {
-					uploadFileOptions[variant] = map[string]string{}
-					uploadFileOptions[variant]["androidManifestPath"] = filepath.Join(mergedManifestsPath, variant, "AndroidManifest.xml")
-					uploadFileOptions[variant]["mappingPath"] = filepath.Join(mergedManifestsPath, "..", "..", "outputs", "mapping", variant, "mapping.txt")
+			mappingPath := filepath.Join(path, "app", "build", "outputs", "mapping")
+
+			if !utils.FileExists(mappingPath) {
+				return fmt.Errorf("unable to find the mapping directory in " + path)
+			}
+
+			if variant == "" {
+				variant, err = android.GetVariant(mappingPath)
+
+				if err != nil {
+					return err
 				}
-			} else {
-				log.Error("unable to find `merged_manifests` in "+path, 1)
 			}
-			//	Path is a file
-		} else {
 
-			if configuration == "" {
-				log.Warn("`--configuration` missing from options for " + path)
-				log.Info("Skipping " + path)
-				continue
+			mappingFile = filepath.Join(mappingPath, variant, "mapping.txt")
+
+			if !utils.FileExists(mappingFile) {
+				return fmt.Errorf(mappingFile + " does not exist on the system")
```

Comment:
As they haven't provided the specific path to the file, this might be confusing. So perhaps something like:

```suggestion
				return fmt.Errorf("unable to find mapping file in the specified project directory")
```